### PR TITLE
Cb 2824 add grouping for connections in multiple mode

### PR DESCRIPTION
**Pull request by tfokina**

### CODE REVIEW

File: webapp/packages/plugin-connections-administration/src/Administration/Connections/ConnectionsTable/ConnectionsTable.tsx [hunk 1]
```diff
@@ -42,6 +43,15 @@ export const ConnectionsTable = observer<Props>(function ConnectionsTable({
   expandedItems,
 }) {
   const translate = useTranslate();
+  const projectService = useService(ProjectsService);
+  const projectsLoader = useResource(ConnectionsTable, ProjectInfoResource, CachedMapAllKey);
```
Comment (devnaumov) [inline webapp/packages/plugin-connections-administration/src/Administration/Connections/ConnectionsTable/ConnectionsTable.tsx:47]: projectsLoader -> projectInfoResource

File: webapp/packages/plugin-connections-administration/src/Administration/Connections/ConnectionsTable/Connection.tsx [hunk 2]
```diff
@@ -40,9 +40,10 @@ const styles = css`
 interface Props {
   connectionKey: IConnectionInfoParams;
   connection: DatabaseConnection;
+  projectName?: string | null;
```
Comment (devnaumov) [inline webapp/packages/plugin-connections-administration/src/Administration/Connections/ConnectionsTable/Connection.tsx:43]: do we really need undefined and null at the same time?
better to keep undefined and remove the null

File: webapp/packages/plugin-connections-administration/src/Administration/Connections/ConnectionsTable/Connection.tsx [hunk 3]
```diff
@@ -61,6 +62,8 @@ export const Connection = observer<Props>(function Connection({ connectionKey, c
       <TableColumnValue title={connection.name} expand ellipsis>{connection.name}</TableColumnValue>
       <TableColumnValue>{connection.host}{connection.host && connection.port && `:${connection.port}`}</TableColumnValue>
       <TableColumnValue>{connection.folder && connection.folder}</TableColumnValue>
+      {projectName !== undefined && (
```
Comment (devnaumov) [inline webapp/packages/plugin-connections-administration/src/Administration/Connections/ConnectionsTable/Connection.tsx:65]: projectName &&  < />
and then projectName ?? '' can be omitted

**Review by devnaumov — CHANGES_REQUESTED** [review]

File: webapp/packages/plugin-connections-administration/src/Administration/Connections/ConnectionsTable/ConnectionsTable.tsx (hunk 1 above)
Comment (Wroud) [reply webapp/packages/plugin-connections-administration/src/Administration/Connections/ConnectionsTable/ConnectionsTable.tsx:47]: it's not exactly resource, it's resource's data loader

File: webapp/packages/plugin-connections-administration/src/Administration/Connections/ConnectionsTable/Connection.tsx (hunk 2 above)
Comment (Wroud) [reply webapp/packages/plugin-connections-administration/src/Administration/Connections/ConnectionsTable/Connection.tsx:43]: we need to detect when we don't have a project name and when we don't display projects column

**Review by Wroud — APPROVED** [review]

File: webapp/packages/plugin-connections-administration/src/Administration/Connections/ConnectionsTable/ConnectionsTable.tsx [hunk 4]
```diff
@@ -42,6 +43,15 @@ export const ConnectionsTable = observer<Props>(function ConnectionsTable({
...
+  const displayProjects = getComputed(() => projectService
+    .activeProjects
+    .filter(project => isGlobalProject(project) || isSharedProject(project)).length > 1);
+
+  function getProjectName(projectId: string) {
```
Comment (devnaumov) [inline webapp/packages/plugin-connections-administration/src/Administration/Connections/ConnectionsTable/ConnectionsTable.tsx:52]: You could make it a variable inside the loop
const projectName = displayProjects ? projectsLoader.resource.get(projectId)?.name : undefined;

**Review by devnaumov — APPROVED** [review]

**Pull request merged**